rocksdb-sys/build.rs

```diff
@@ -1,273 +1,48 @@
-	config.file("rocksdb/utilities/memory/memory_util.cc");
-	config.file("rocksdb/utilities/merge_operators/max.cc");
-	config.file("rocksdb/utilities/merge_operators/put.cc");
-	config.file("rocksdb/utilities/merge_operators/string_append/stringappend.cc");
-	config.file("rocksdb/utilities/merge_operators/string_append/stringappend2.cc");
-	config.file("rocksdb/utilities/merge_operators/uint64add.cc");
-	config.file("rocksdb/utilities/option_change_migration/option_change_migration.cc");
-	config.file("rocksdb/utilities/options/options_util.cc");
-	config.file("rocksdb/utilities/persistent_cache/block_cache_tier.cc");
-	config.file("rocksdb/utilities/persistent_cache/block_cache_tier_file.cc");
-	config.file("rocksdb/utilities/persistent_cache/block_cache_tier_metadata.cc");
-	config.file("rocksdb/utilities/persistent_cache/persistent_cache_tier.cc");
-	config.file("rocksdb/utilities/persistent_cache/volatile_tier_impl.cc");
-	config.file("rocksdb/utilities/redis/redis_lists.cc");
-	config.file("rocksdb/utilities/simulator_cache/sim_cache.cc");
-	config.file("rocksdb/utilities/spatialdb/spatial_db.cc");
-	config.file("rocksdb/utilities/table_properties_collectors/compact_on_deletion_collector.cc");
-	config.file("rocksdb/utilities/transactions/optimistic_transaction.cc");
-	config.file("rocksdb/utilities/transactions/optimistic_transaction_db_impl.cc");
-	config.file("rocksdb/utilities/transactions/pessimistic_transaction.cc");
-	config.file("rocksdb/utilities/transactions/pessimistic_transaction_db.cc");
-	config.file("rocksdb/utilities/transactions/transaction_base.cc");
-	config.file("rocksdb/utilities/transactions/transaction_db_mutex_impl.cc");
-	config.file("rocksdb/utilities/transactions/transaction_lock_mgr.cc");
-	config.file("rocksdb/utilities/transactions/transaction_util.cc");
-	config.file("rocksdb/utilities/transactions/write_prepared_txn.cc");
-	config.file("rocksdb/utilities/ttl/db_ttl_impl.cc");
-	config.file("rocksdb/utilities/write_batch_with_index/write_batch_with_index.cc");
-	config.file("rocksdb/utilities/write_batch_with_index/write_batch_with_index_internal.cc");
+    let mut build = out.join("build");
 
-	config.file("build_version.cc");
+    if cfg!(target_os = "windows") {
+        let profile = match &*env::var("PROFILE").unwrap_or("debug".to_owned()) {
+            "bench" | "release" => "Release",
+            _ => "Debug",
+        };
+        build = build.join(profile);
+    }
 
-	config.cpp(true);
-	config.compile("librocksdb.a");
+    println!("cargo:rustc-link-search=native={}", build.display());
+    println!("cargo:rustc-link-lib=static=rocksdb");
+    println!("cargo:rustc-link-lib=static=snappy");
 
-	println!("cargo:rustc-link-lib=static=snappy");
+    // https://github.com/alexcrichton/cc-rs/blob/ca70fd32c10f8cea805700e944f3a8d1f97d96d4/src/lib.rs#L891
+    if cfg!(any(target_os = "macos", target_os = "freebsd", target_os = "openbsd")) {
```

Comment:
These lines are a bit harmful, see https://github.com/paritytech/parity-ethereum/issues/9259.
But I don't think it's a valid reason to delay this PR. We can just apply the same fix as for `snappy-sys` in the future.